query-graphs-js/src/querygraph.ts

```diff
@@ -807,67 +807,53 @@ function federateSubgraphs(supergraph: Schema, subgraphs: QueryGraph[]): QueryGr
    * override condition of `true`.
    */
-  const allOverrideApplicationsWithLabel = [];
-  for (const [i, subgraph] of subgraphs.entries()) {
+  const subgraphsByName = new Map(subgraphs.map((s) => [s.name, s]));
+  for (const [i, toSubgraph] of subgraphs.entries()) {
     const subgraphSchema = schemas[i];
     const subgraphMetadata = federationMetadata(subgraphSchema);
     assert(subgraphMetadata, `Subgraph ${i} is not a valid federation subgraph`);
-    const overrideDirectiveApplications = subgraphMetadata.overrideDirective().applications();
-    allOverrideApplicationsWithLabel.push(
-      ...overrideDirectiveApplications.filter(a => a.arguments().label).map(a => [a, subgraph.name] as [Directive, string])
-    );
-  }
-
-  for (const application of allOverrideApplicationsWithLabel) {
-    const [overrideApplication, toSubgraphName] = application;
-    const label = overrideApplication.arguments().label;
-    const fromSubgraphName = overrideApplication.arguments().from;
-
-    const toSubgraphIndex = subgraphs.findIndex(s => s.name === toSubgraphName);
-    const toSubgraph = subgraphs[toSubgraphIndex];
-    const fromSubgraphIndex = subgraphs.findIndex(s => s.name === fromSubgraphName);
-    const fromSubgraph = subgraphs[fromSubgraphIndex];
-    assert(toSubgraph && fromSubgraph, () => `Subgraph ${toSubgraphName} or ${fromSubgraphName} not found`);
-
-    const field = overrideApplication.parent;
-    assert(field instanceof NamedSchemaElement, () => `@override should have been on a field, got ${field}`)
-    const type = field.parent;
-
-    const [toSubgraphVertex, ...unexpectedAdditionalToVertices] = toSubgraph.verticesForType(type.name);
-    assert(toSubgraphVertex && unexpectedAdditionalToVertices.length === 0, () => `Subgraph ${toSubgraphName} should have exactly one vertex for type ${type.name}`);
-    const [fromSubgraphVertex, ...unexpectedAdditionalFromVertices] = fromSubgraph.verticesForType(type.name);
-    assert(fromSubgraphVertex && unexpectedAdditionalFromVertices.length === 0, () => `Subgraph ${fromSubgraphName} should have exactly one vertex for type ${type.name}`);
-
-    const toSubgraphEdges = toSubgraph.outEdges(toSubgraphVertex);
-    for (const edge of toSubgraphEdges) {
-      if (
-        edge.transition.kind === "FieldCollection"
-        && edge.transition.definition.name === field.name
-      ) {
-        const head = copyPointers[toSubgraphIndex].copiedVertex(toSubgraphVertex);
-        const copiedEdge = builder.edge(head, edge.index);
-
-        copiedEdge.overrideCondition = {
-          label,
-          condition: true,
+    const overrideDirectiveApplicationsWithLabel =
+      subgraphMetadata
+        .overrideDirective()
+        .applications()
+        .filter(a => a.arguments().label);
+
+    for (const application of overrideDirectiveApplicationsWithLabel) {
+      const { from, label } = application.arguments();
+      assert(label, () => `Expected @override application to have a label`);
```

Comment:
nit: This `for` loop could be over `subgraphMetadata.overrideDirective().applications()`, and this `assert` could then be replaced with `if (!label) { continue; }`

Reply:
[f08a347](https://github.com/apollographql/federation/pull/2902/commits/f08a347fd05b125d5f3ca54531a7c2f4ebe130c4)